### ISSUE TITLE
Run `dbt deps` when using `load_mode=LoadMode.DBT_LS`

### ISSUE THREAD

**Issue by tatiana** [astronomer/astronomer-cosmos#445]
As of Cosmos 1.0, when we run `dbt ls` as part of `load_mode=LoadMode.DBT_LS`, we don't pre-run `dbt deps`.

This would be a useful feature based on the community feedback, including this thread in the Airflow Community Slack.
https://apache-airflow.slack.com/archives/C059CC42E9W/p1691413162622129?thread_ts=1691398106.912769&cid=C059CC42E9W

The goal with this ticket is to run `dbt deps` before `dbt ls`.

**Comment by DanMawdsleyBA** [comment]
@tatiana This is my first change here so forgive me for any mistakes. I made the changes in a forked repo tested it locally and ran ok. I also added a package.yml to the example dev project so that should prove the change works when it runs the exitisting tests. See below screenshot of logs when I ran:
![image](https://github.com/astronomer/astronomer-cosmos/assets/134394384/ea2e66f4-dcaf-4c13-a6cb-d218a5c9b94d)
I looked at the tests that ran a couple of failures on the integration tests although I don't think it is related to the change I made:
![image](https://github.com/astronomer/astronomer-cosmos/assets/134394384/9e0bd152-fdcf-4291-9e93-aff68049556c)
All the failures were the same. So let me know if you're happy with the changes :)
